interfaces/tempo_cluster/v0/schema.py

```diff
@@ -5,30 +5,36 @@
 - RequirerSchema
 """
 from enum import Enum
-from typing import Optional, Dict, Any, Literal
+from typing import Optional, Dict, Any, Literal, MutableMapping, List
+
+import json
 
 from interface_tester.schema_base import DataBagSchema
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, ConfigDict, Field, ValidationError, Json
 
 
 class TempoClusterProviderAppData(BaseModel):
     """TempoClusterProviderAppData."""
-    tempo_config: str = Field(
+    worker_config: Json[str] = Field(
         description="The tempo configuration that the requirer should run with."
                     "Yaml-encoded. Must conform to the schema that the presently deployed "
                     "workload version supports; for example see: "
                     "https://grafana.com/docs/tempo/latest/configuration/#configure-tempo."
     )
-    loki_endpoints: Optional[Dict[str, str]] = Field(
+    loki_endpoints: Optional[Json[Dict[str, str]]] = Field(
         default=None,
         description="List of loki-push-api endpoints to which the worker node can push any logs it generates.")
-    ca_cert: Optional[str] = Field(default=None, description="CA certificate for tls encryption.")
-    server_cert: Optional[str] = Field(default=None, description="Server certificate for tls encryption.")
-    privkey_secret_id: Optional[str] = Field(
+    ca_cert: Optional[Json[str]] = Field(default=None, description="CA certificate for tls encryption.")
+    server_cert: Optional[Json[str]] = Field(default=None, description="Server certificate for tls encryption.")
+    privkey_secret_id: Optional[Json[str]] = Field(
```

Comment:
What's the benefit of use JSON for these rather than just plain strings? (If that's documented elsewhere, please just point me to it and I'll go read it there to understand).

Reply:
This is a convention that we use across all our interfaces: we are json-encoding the contents of the databags. You can see a sample implementation [in the shared worker-coordinator object](https://github.com/canonical/cos-lib/blob/main/src/cosl/coordinated_workers/interface.py#L58).

You can see that ingress is following the same approach [here](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/ingress/v2/schema.py), same with [does prometheus_scrape too](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/prometheus_scrape/v0/schema.py#L128).